ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/NormalizeAttributes.java

```diff
@@ -53,20 +53,30 @@ public static String normalizeChannel(String name) {
    * Normalize an operating system string to one of the three major desktop platforms,
    * one of the two major mobile platforms, or "Other".
    *
-   * <p>Reimplements and combines the logic of {@code os} and {@code mobile_os} from
+   * <p>Previously implemented by {@code os} and {@code mobile_os} from
    * https://github.com/mozilla-services/lua_sandbox_extensions/blob/607b3c7b9def600ddbd373a14f67a55e1c69f7a1/moz_telemetry/modules/moz_telemetry/normalize.lua#L184-L215
    */
-  public static String normalizeOs(String name) {
-    if (name.startsWith("Windows") || name.startsWith("WINNT")) {
-      return WINDOWS;
-    } else if (name.startsWith("Darwin")) {
-      return MAC;
-    } else if (name.contains("Linux") || name.contains("BSD") || name.contains("SunOS")) {
-      return LINUX;
-    } else if (name.startsWith("iOS") || name.contains("iPhone")) {
+  public static String normalizeOs(String name, Optional<String> normalizedAppName) {
+    boolean appIsNotDesktop = normalizedAppName //
+        .map(v -> !OTHER.equals(v) && !FIREFOX.equals(v)).orElse(false);
+    // Don't look for desktop os names if app is not desktop
+    if (!appIsNotDesktop) {
```

Comment:
There are logically two "not"s here, which we should avoid. I think perhaps we invert the boolean and call it `mightBeDesktop`, so that this condition becomes `if (mightBeDesktop)`.